lib/pointer.js

```diff
@@ -189,6 +189,25 @@ Pointer.parse = function (path, originalPath) {
   return pointer.slice(1);
 };
 
+
+/**
+ * Like `Pointer.parse` but only returns the length of the pointer.
+ * Same as `Pointer.parse(path).length` but faster.
+ * @param path
+ * @returns {number}
+ */
+Pointer.parseLength = function (path) {
+  // Get the JSON pointer from the path's hash
+  let pointer = url.getHash(path).substr(1);
+
+  // If there's no pointer, then there are no tokens
+  if (!pointer) {
+    return 0;
+  }
+
+  return (pointer.match(/\//g) || []).length;
```

Comment:
Since we're here for speed, I suppose that

```
let lastIndex = -1;
let i = 0;
while ((lastIndex = pointer.indexOf('/', lastIndex + 1)) !== -1) i++;
return i;
```
might be considerably faster, but if we spend, say, 4ms on `parseLength` calls then the diff would be like 1-2ms, meaning there's no point in changing anything.